src/device/pq.rs

```diff
@@ -0,0 +1,94 @@
+use std::{net::SocketAddr, sync::Arc, time::Duration};
```

Comment:
IDK, I've placed it here because it's the device that maintains the connection logic. Maybe doing a separate crate for PQ? I think I will wait for more input from the reviewer because it's quite subjective IMHO

Reply:
I would also suggest to move this into a separate crate with all the PQ logic.

Reply:
Ok, let me do it then

Reply:
Done